pandas/_libs/groupby.pyx

```diff
@@ -704,21 +733,24 @@ def group_mean(floating[:, ::1] out,
             counts[lab] += 1
             for j in range(K):
                 val = values[i, j]
-                # not nan
-                if val == val:
-                    nobs[lab, j] += 1
-                    y = val - compensation[lab, j]
-                    t = sumx[lab, j] + y
-                    compensation[lab, j] = t - sumx[lab, j] - y
-                    sumx[lab, j] = t
-
+                if val !=val or (is_datetimelike and val == NPY_NAT):
```

Comment:
whitespace after "!="

can use treat_as_na?

Reply:
Done for the first part. Thanks.

On the second part: Good call! I originally reused `treat_as_na` in this PR, however, I found that it introduced a side-effect in the other usage of `treat_as_na` in `cummin_max` since, although `_treat_as_na` accepts `rank_t` as its value argument, it's conditions are currently specifically targeting int64_t values. Hence, I needed to add a condition for `floating` values.

This was made more complicated since:
 *  cummin_max uses `groupby_t` as the data_type for its value array 
 * `group_mean` uses `floating` as the data_type for its value array  
 * and  _treat_as_na uses`rank_t` 

I wasn't sure how these union types behave in practice and had problems getting tests to pass so decided to go for this option.